lib/hotel_front_desk.rb

```diff
@@ -0,0 +1,283 @@
+    
+    if results == []
+      string = "\nNO RESERVATIONS FOR DATE #{date}" 
+    else
+      string = "\nLISTING RESERVATIONS FOR DATE #{date}..."
+      results.each { |reservation| string << "\n  #{reservation}" }
+    end
+    return string
+  end
+  
+  def list_available_rooms(date_range)
+    results = find_all_avail_rooms(date_range)
+    if results == []
+      string = "\nNO ROOMS AVAILABLE FOR #{date_range.start_date} TO #{date_range.end_date}"
+    else
+      string = "\nLISTING AVAILABLE ROOMS FOR #{date_range.start_date} TO #{date_range.end_date}..."
+      results.each { |room| string << "\n  #{room}"}
+    end
+    return string
+  end
+  
+  def get_room_from_id(id_arg)
+    if id_arg.class != Integer
+      raise ArgumentError, "Room id# should be an integer..."
+    end
+    
+    room = @all_rooms.find { |room| room.id == id_arg }
+    
+    if room
+      return room
+    else
+      raise ArgumentError, "Room ##{id_arg} does not exist"
+    end
+  end
+  
+  def get_rooms_from_ids (room_ids)
+    # given room_ids in an array, return an array of Room instances
+    
+    if room_ids.length > num_rooms_in_hotel
+      raise ArgumentError, "You're asking for more rooms than in existence at this here hotel"
+    elsif room_ids.uniq.length != room_ids.length
+      raise ArgumentError, "Some of your args are duplicates, fix it plz"
+    end
+    
+    rooms = []
+    (room_ids).each do |id|
+      room = get_room_from_id(id)
+      rooms << room
+    end
+    return rooms
+  end
+
+  def get_block_from_id(id_arg)
+    if id_arg.class != Integer
+      raise ArgumentError, "Block id# should be an integer..."
+    end
+    
+    block = @all_blocks.find { |block| block.id == id_arg }
+    
+    if block
+      return block
+    else
+      raise ArgumentError, "Block ##{id_arg} does not exist"
+    end
+  end
+  
+  def make_block(date_range:, room_ids:, new_nightly_rate:)
+    # Validate date_range
+    if date_range.class != Date_range
+      raise ArgumentError, "Must pass in a Date_range object"
+    else
+      @date_range = date_range
+    end
+    
+    # Validate room_ids[]
+    if room_ids.class != Array
+      raise ArgumentError, "Require room_ids to be in an array"
+    elsif room_ids.length == 0
+      raise ArgumentError, "You didn't put anything in room_ids"
+    elsif room_ids.length > MAX_BLOCK_SIZE
+      raise ArgumentError, "Max block size allowed is #{MAX_BLOCK_SIZE}"
+    else
+      @room_ids = room_ids
+      # continue validating as we check rooms' availability later 
+    end
+    
+    # Validate new_nightly_rate
+    if !non_zero_integer?(new_nightly_rate)
+      raise ArgumentError, "We need a non-zero Integer for new_nightly_rate"
+    elsif new_nightly_rate >= STANDARD_RATE
+      raise ArgumentError, "Shouldn't the new_nightly_rate be a discount? compared to standard rate of #{STANDARD_RATE}?"
+    else
+      @new_nightly_rate = new_nightly_rate
```

Comment:
Did you accidentally assign this? This is adding @new_nightly_rate as an instance variable to Hotel_front_desk.

Reply:
haha yeah line 192 and 193 are mistakes, assigned to a HotelFrontDesk variable that I didn't even use.  Lines are now deleted and everything still tested 100%.